pkg/tfvars/openstack/openstack.go

```diff
@@ -68,6 +69,10 @@ func TFVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, cloud string, exter
 		MasterAvailabilityZones: zones,
 	}
 
+	if mpool != nil && mpool.RootVolume != nil {
+		cfg.MasterRootVolumeAvalabilityZones = mpool.RootVolume.Zones
```

Comment:
The `mpool` passed into this function is `ControlPlane.Platform.OpenStack`. It does not consider whether the zones may be set in `Openstack.DefaultMachinePlatform`.

Reply:
yeah, now it's fixed